nextcord/interactions.py

```diff
@@ -559,6 +539,485 @@ async def send(
+            Union[SlashApplicationSubcommand, BaseApplicationCommand]
+        ] = None
+
+    def _from_data(self, data: InteractionPayload):
+        super()._from_data(data=data)
+
+        self.app_command_name: str = self.data["name"]  # type: ignore # self.data should be present here
+        self.app_command_id: int = self.data["id"]  # type: ignore # self.data should be present here
+
+        try:
+            self.options: list = self._get_application_options(self.data["options"])  # type: ignore # self.data should be present here
+        except KeyError:
+            self.options: list = []
+
+    def _set_application_command(
+        self, app_cmd: Union[SlashApplicationSubcommand, BaseApplicationCommand]
+    ):
+        self.application_command = app_cmd
+
+    def _get_application_options(self, data):
+        options_collection: list = []
+
+        for option in data:
+            if "focused" in option:
+                self.focused_option = SlashOptionData(option)
+                options_collection.append(self.focused_option)
+                continue
+
+            options_collection.append(SlashOptionData(option))
+
+        return options_collection
+
+
+class ViewInteraction(Interaction):
+    """Represents the base class for all Discord interactions.
+
+    .. container:: operations
+
+        .. describe:: x == y
+
+            Checks if two interactions are equal.
+
+        .. describe:: x != y
+
+            Checks if two interactions are not equal.
+
+        .. describe:: hash(x)
+
+            Returns the interaction's hash.
+
+    .. versionadded:: 2.0
+
+    .. versionchanged:: 2.1
+        :class:`Interaction` is now hashable.
+
+    Attributes
+    ----------
+    id: :class:`int`
+        The interaction's ID.
+    type: :class:`InteractionType`
+        The interaction type.
+    guild_id: Optional[:class:`int`]
+        The guild ID the interaction was sent from.
+    channel_id: Optional[:class:`int`]
+        The channel ID the interaction was sent from.
+    locale: Optional[:class:`str`]
+        The users locale.
+    guild_locale: Optional[:class:`str`]
+        The guilds preferred locale, if invoked in a guild.
+    application_id: :class:`int`
+        The application ID that the interaction was for.
+    user: Optional[Union[:class:`User`, :class:`Member`]]
+        The user or member that sent the interaction.
+    token: :class:`str`
+        The token to continue the interaction. These are valid
+        for 15 minutes.
+    data: :class:`dict`
+        The raw View specific interaction data.
+    attached: :class:`InteractionAttached`
+        The attached data of the interaction. This is used to store any data you may need inside the interaction for convenience. This data will stay on the interaction, even after a :meth:`Interaction.application_command_before_invoke`.
+    message: :class:`Message`
+        The message the component is attached to
+    component_id: :class:`int`
```

Comment:
It looks like the component id attribute is really the custom id. Custom ids for components are typed to be strings, not integers.